doc/content/hpc_build_tips.md

```diff
@@ -0,0 +1,156 @@
+- `module reset`, restores your shell session to the HPC system default modules.
+- `module purge`, unloads all currently loaded modules. Some modules are mandatory and so Lmod will refuse to unload them.
+
+Before you start loading modules, it's best to consult the minimum requirements of MOOSE, NekRS, OpenMC, and DAGMC:
+
+1. MOOSE: [https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html](https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html)
+2. NekRS: [https://github.com/Nek5000/nekRS?tab=readme-ov-file#build-instructions](https://github.com/Nek5000/nekRS?tab=readme-ov-file#build-instructions)
+3. OpenMC: [https://docs.openmc.org/en/stable/usersguide/install.html#prerequisites](https://docs.openmc.org/en/stable/usersguide/install.html#prerequisites)
+4. DAGMC: [https://svalinn.github.io/DAGMC/install/dependencies.html](https://svalinn.github.io/DAGMC/install/dependencies.html)
+
+Out of the above requirements, MOOSE's are the strictest. At present the MOOSE minimum requirements are:
+
+- A "modern" C/C++ compiler: GCC 8.5.0 - 12.2.1 or LLVM/Clang 10.0.1 - 16.0.6;
+- MPI 3.1 or newer (either OpenMPI or MPIH);
+- OpenMP;
+- Python 3.9-3.11 with `packaging` and `pyaml`. MOOSE also lists `jinja2` as a mandatory dependency, but this is not required to build or run MOOSE applications.
+
+From here, you will need to find a set of modules that satisfy these minimum requirements. This tends to involve a substantial amount of trial and error which depends a lot on what each HPC system offers. It's best to consult the documentation for your specific HPC system to see which modules they recommend for specific requirements of the MOOSE software stack.
+
+Once you've loaded a set of modules that you believe will compile Cardinal, you need to set up a collection of environment variables. Assuming your Cardinal installation is located in a folder called `projects` in your home directory, they are:
+
+```bash
+# These are hints for the compiler that should be used while building.
+export CC=mpicc
+export CXX=mpicxx
+export FC=mpif90
+
+# This lets the build scripts know that you want all of Cardinal's optional dependencies.
+export ENABLE_NEK=yes
+export ENABLE_OPENMC=yes
+export ENABLE_DAGMC=yes
+
+# This points to the actual location where your Cardinal directory is stored on the HPC's file system.
+# Most HPC systems use symbolic links for directories and files to hide their true paths (to make life
+# easier for users), which is why this is necessary.
+DIRECTORY_WHERE_YOU_HAVE_CARDINAL=$HOME/projects
+HOME_DIRECTORY_SYM_LINK=$(realpath -P $DIRECTORY_WHERE_YOU_HAVE_CARDINAL)
+export NEKRS_HOME=$HOME_DIRECTORY_SYM_LINK/cardinal/install
+
+# These environment variables indicate how many processes you want to build with. Be careful with the number
+# of jobs specified, most HPC systems limit the number of processes you can run with on login nodes.
```

Comment:
Can we add a comment about memory (during compilation) increasing when you build in parallel? I think some people have run into this in the past, and were able to fix it by reducing these numbers.